plugins/woocommerce/src/Database/Migrations/CustomOrderTable/PostsToOrdersMigrationController.php

```diff
@@ -71,9 +72,101 @@ public function get_migrated_meta_keys() {
 	public function migrate_orders( array $order_post_ids ): void {
 		$this->error_logger = WC()->call_function( 'wc_get_logger' );
 
+		$using_transactions = $this->maybe_start_transaction();
+		if ( null === $using_transactions ) {
+			return;
+		}
+
+		$errors_were_logged = false;
+
 		foreach ( $this->all_migrators as $migrator ) {
-			$this->do_orders_migration_step( $migrator, $order_post_ids );
+			$errors_were_logged = $this->do_orders_migration_step( $migrator, $order_post_ids );
+			if ( $errors_were_logged && $using_transactions ) {
+				$this->rollback_transaction();
```

Comment:
Should we take any action if the rollback fails (suppose the 'db has gone away')? I'm sure that would be unusual, but it's possible.

Reply:
It's my understanding that any transaction that is not explicitly commited gets rolled back sooner or later, either explicitly by a `ROLLBACK` command, or when the connection is closed. Any failure here will be logged though, I think that should be enough.